e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -667,6 +667,304 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
+                Assert.True(response.IsSuccessful, "DELETE /registry/v2/application/{applicationId} failed!");
+            }
+        }
+
+        /// <summary>
+        /// Gets endpoints from registry
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        public static async Task Registry_ActivateEndpointAsync(IIoTPlatformTestContext context, CancellationToken ct = default) {
+            var accessToken = await GetTokenAsync(context, ct).ConfigureAwait(false);
+
+            Assert.False(string.IsNullOrWhiteSpace(context.OpcUaEndpointId), "Endpoint not set in the test context");
+
+            var client = new RestClient(context.IIoTPlatformConfigHubConfig.BaseUrl) {
+                Timeout = TestConstants.DefaultTimeoutInMilliseconds
+            };
+
+            var request = new RestRequest(Method.POST);
+            request.AddHeader(TestConstants.HttpHeaderNames.Authorization, accessToken);
+            request.Resource = string.Format(TestConstants.APIRoutes.RegistryActivateEndpointsFormat, context.OpcUaEndpointId);
+
+            var response = client.ExecuteAsync(request, ct).GetAwaiter().GetResult();
+            Assert.NotNull(response);
+
+            if (!response.IsSuccessful) {
+                context.OutputHelper.WriteLine($"StatusCode: {response.StatusCode}");
+                context.OutputHelper.WriteLine($"ErrorMessage: {response.ErrorMessage}");
+                Assert.True(response.IsSuccessful, "POST /registry/v2/endpoints/{endpointId}/activate failed!");
+            }
+
+            Assert.Empty(response.Content);
+
+            while (true) {
+                Assert.False(ct.IsCancellationRequested, "Endpoint was not activated within the expected timeout");
+
+                var endpointList = await Registry_GetEndpointsAsync(context).ConfigureAwait(false);
+                var endpoint = endpointList.FirstOrDefault(e => string.Equals(e.Id, context.OpcUaEndpointId));
+
+                if (string.Equals(endpoint.ActivationState, TestConstants.StateConstants.ActivatedAndConnected)
+                        && string.Equals(endpoint.EndpointState, TestConstants.StateConstants.Ready)) {
+                    return;
+                }
+
+                context.OutputHelper.WriteLine(string.IsNullOrEmpty(endpoint.Url) ? "Endpoint not found" :
+                    $"Endpoint state: {endpoint.EndpointState}, activation: {endpoint.ActivationState}");
+
+                await Task.Delay(TestConstants.DefaultDelayMilliseconds).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Gets endpoints from registry
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        public static async Task<List<(string Id, string Url, string ActivationState, string EndpointState)>> Registry_GetEndpointsAsync(
```

Comment:
This method does the same as WaitForEndpointDiscoveryToBeCompleted when requestedEndpointUrls=null

Reply:
Not completely, but I extracted the shared functionality into a private method